### PR TITLE
Ct 3924 make sar ir cases triggered

### PR DESCRIPTION
**Pull request by willmcb**
## Description
Makes SAR IR automatically a triggered case. This also re-adds a separate workflow for SAR IR as this we agreed it differs enough to justify having a separate one. This PR also removed the opt out guard clause for SAR IR cases in the `#can_be_unflagged_for_clearance?` method as calls to it are removed in the new workflow.

Also moves the `steps_are_complete?` method to Steppable just to try to start moving things out of the controller.

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [ ] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [ ] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [ ] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed

### Related JIRA tickets
Part of [CT-3924](https://dsdmoj.atlassian.net/browse/CT-3924)

### Manual testing instructions
create a new SAR IR case - it should now appear as triggered on the show page

### CODE REVIEW

**Review by ymao2 — COMMENTED** [review]
Nice work and movement to have workflow specifically for SAR IR 👍 ,  but I have some questions in the my comment, happy to discuss it more.

File: config/state_machine/configs/00_moj/60_sar_internal_review/10_sar_internal_review_preamble.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#########################################################################
+#                                                                       #
+#                         SAR INTERNAL REVIEW                           #
+#                                                                       #
+#########################################################################
+  sar_internal_review:
+    name: Subject Access Request Internal Review
+    permitted_workflows:
+      - standard
+      - trigger
```
[the remark this answers is not in the document]
Comment (willmcb) [reply config/state_machine/configs/00_moj/60_sar_internal_review/10_sar_internal_review_preamble.yml:10]: Yep I agree! 👍

File: config/state_machine/configs/00_moj/60_sar_internal_review/30_sar_internal_review_trigger_workflow.yml [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+                add_message_to_case:
+                  after_transition: Workflows::Hooks#notify_responder_message_received
+                assign_to_new_team:
+                  transition_to: awaiting_responder
+                  after_transition: Workflows::Hooks#assign_responder_email
+                destroy_case:
+                edit_case:
+                  if: Workflows::Predicates#overturned_editing_enabled?
+                extend_sar_deadline:
+                flag_for_clearance:
```
[the remark this answers is not in the document]
Comment (willmcb) [reply config/state_machine/configs/00_moj/60_sar_internal_review/30_sar_internal_review_trigger_workflow.yml:43]: Yep its not needed I'll remove 👍

File: app/controllers/cases/sar_internal_review_controller.rb [hunk 3]
```diff
@@ -53,17 +53,20 @@ def create_case
       session[session_state] = nil
       flash[:creating_case] = true
       flash[:notice] = "Case created successfully"
+      make_case_a_triggered_case
```
[the remark this answers is not in the document]
Comment (willmcb) [reply app/controllers/cases/sar_internal_review_controller.rb:95]: I have refactored to use the CaseCreateService and a new Builder class now 👍 Please let me know your thoughts.

File: app/models/builders/stepped_case_builder.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          @ready_for_creation = true
+        end
+      end
+    end
+
+    def session_state
+      "#{@case_type.type_abbreviation.downcase}_state"
+    end
+  end
+end
```
Comment (ymao2) [inline app/models/builders/stepped_case_builder.rb:56]: good idea to have separate module to handle the session data

**Review by ymao2 — APPROVED** [review]
Thank you for taking time to do those changes based on the feedback, Nice work!

**Pull request merged**